src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/IdentToken.cs

```diff
@@ -70,7 +71,18 @@ internal override Token Clone(Span ts)
         // REVIEW ragru: having a property for every possible error isn't scalable.
         internal bool HasDelimiters => HasDelimiterStart;
 
-        internal bool HasErrors => IsModified || (HasDelimiterStart && !HasDelimiterEnd);
+        /// <summary>
+        /// Whether an identifier has errors.
+        /// </summary>
+        public bool HasErrors => IsModified || (HasDelimiterStart && !HasDelimiterEnd);
+
+        // TODO: How to do this properly?
+        public static string MakeValidIdentifier(string value)
```

Comment:
>MakeValidIdentifier [](http://example.com/codeflow?start=29&length=19)

add a unit test.  (Theory + InlineData)

Reply:
Consider null, empty string and a set of whitespaces

Reply:
Added some tests, let me know if we need to be more thorough.